17_2_chart.js

```diff
@@ -0,0 +1,34 @@
+const results = [
+    { name: 'Satisfied', count: 1043, color: 'lightblue' },
+    { name: 'Neutral', count: 563, color: 'lightgreen' },
+    { name: 'Unsatisfied', count: 510, color: 'pink' },
+    { name: 'No comment', count: 175, color: 'silver' },
+];
+
+const ctx = document.querySelector('canvas').getContext('2d');
+const total = results.reduce((sum, { count }) => sum + count, 0);
+let currentAngle = -0.5 * Math.PI;
+const centerX = 300;
+const centerY = 150;
+
+// Add code to draw the slice labels in this loop.
+for (let result of results) {
```

Comment:
Please, try to group expressions in that loop. Also, maybe you will be able to extract some parts into separate functions then.